### ISSUE TITLE
Incorrect function name reported in a warning message

### ISSUE THREAD

**Issue by BarryBo** [FStarLang/kremlin#16]
The repro is to run the .json file I emailed you, like this:

```
Kremlin.native sha256_main.i.json -warn-error +1..4 -skip-compilation -add-include \"DafnyLib.h\"
Warning 4: in declaration sha256_main_i.SHA256_DigestOneBlock, in the definition of sha256_main_i: Malformed input:
type mismatch, uint64_t (a.k.a. uint64_t) vs Dafny_BigInt (a.k.a. Dafny_BigInt)
Dropping sha256_main_i (at checking time)
```
Note the warning says "in the definition of sha256_main_i".  I think it is reporting the namespace of the function definition, not the function name, which is SHA256_ComputeInitialWs.

**Comment by msprotz** [comment]
Alright, so the bug report is about the location of the error, not that the error itself is spurious? (Just making sure I'm not mistaken.)

**Comment by msprotz** [comment]
Ok, so the location was roughly correct, but it seemingly made no sense because this was an error introduced by KreMLin itself when performing rewritings.

I added a little message on the output to help tell apart an error because of Kremlin's internal rewritings vs an error in the input file... and I'm looking right now at the actual cause of the error.

**Comment by msprotz** [comment]
Fixed by 5a2f9ad
